src/bin/inx-chronicle/cli.rs

```diff
@@ -255,55 +256,37 @@ impl ClArgs {
-                                        db.update_all_analytics(index, prev_analytics).await?;
-                                        influx_db
-                                            .insert_all_analytics(timestamp, index, prev_analytics.clone())
-                                            .await?;
-                                    } else {
-                                        let analytics = db.get_all_analytics(index).await?;
-                                        prev_analytics = Some(analytics.clone());
-                                        influx_db.insert_all_analytics(timestamp, index, analytics).await?;
-                                    }
 
-                                    #[cfg(feature = "metrics")]
-                                    {
-                                        let elapsed = start_time.elapsed();
-                                        influx_db
-                                            .insert(chronicle::db::collections::metrics::AnalyticsMetrics {
-                                                time: chrono::Utc::now(),
-                                                milestone_index: index,
-                                                analytics_time: elapsed.as_millis() as u64,
-                                                chronicle_version: std::env!("CARGO_PKG_VERSION").to_string(),
-                                            })
-                                            .await?;
-                                    }
-                                    tracing::info!("Finished analytics for milestone {}", index);
-                                } else {
-                                    tracing::info!("No milestone in database for index {}", index);
-                                }
+                        let mut handle = shutdown_signal.subscribe();
+                        join_set.spawn(async move {
+                            tokio::select! {
+                                res = fill_analytics(db, influx_db, start_milestone, end_milestone) => res?,
+                                _ = handle.recv() => {},
                             }
                             Ok(())
                         });
                         // Account for inclusive end
                         start_milestone = end_milestone + 1;
                     }
-                    while let Some(res) = join_set.join_next().await {
-                        // Panic: Acceptable risk
-                        res.unwrap()?;
+
+                    // We wait for either the interrupt signal or the completion of filling the analytics.
+                    loop {
+                        tokio::select! {
+                            _ = crate::process::interrupt_or_terminate() => {
+                                tracing::info!("received ctrl-c or terminate");
+                                shutdown_signal.send(())?;
```

Comment:
This won't await the tasks after a `ctrl-c` FYI, so they will likely still be aborted by the end of the run.

Reply:
yeah, I thought about that... but does it really matter? I can mitigate this problem by just starting at this index again, right?

Reply:
Well, it kind of defeats the entire purpose of this PR lol

Reply:
Then I must have misunderstood what you meant with the term "run" (I thought you mean a run for a single milestone). The purpose of this PR is to abort all tasks spawned onto the `JoinSet` and it does just that (tested).

Reply:
The outer loop prevents the issue I was describing, which I missed. However, I do not like that the select is inside that loop. Sure, the difference probably wouldn't be noticed with regular usage, but consider that this is possible:
1. 10 tasks are spawned.
2. Tasks are canceled with ctrl-c.
3. Task 1 is joined.
4. Ctrl-c is hit again before any other tasks are complete.

In this situation, having pressed `ctrl-c` twice, it should force quit. However, it is possible if pressed quickly to instead complete the `interrupt_or_terminate` future multiple times.

Reply:
Pushed another solution that is as close at it can be to what I think you want.